vllm/timings/timer.py

```diff
@@ -0,0 +1,75 @@
+import time
+from contextlib import contextmanager
+from threading import RLock
+
+import numpy
+
+__all__ = ["Timer"]
+
+
+class Timer:
+    """Timer to log timings during inference requests. Should be used through 
+        timings.utils.get_singleton_manager()
+    """
+    _instance = None
+
+    def __init__(self, avg_after_iterations: int, log: bool):
+        """
+        :param avg_after_iterations: the number of iterations that have to occur
+            before the time measurements are averaged and displayed to the user
+        :param log: whether or not time logging is enabled
+        """
+        self.log = log
```

Comment:
log is bool so id rename to something similar to is_log_enabled above

Reply:
+1